### ISSUE TITLE
Land on 'if (FALSE)' for unreachable code?

### ISSUE THREAD

**Issue by MichaelChirico** [r-lib/lintr#2260]
```
lint("
if (FALSE) {
  foo()
}",
unreachable_code_linter())
# <text>:3:3: warning: [unreachable_code_linter] Code inside a conditional loop with a deterministically false condition should be removed.
#   foo()
#   ^~~~~
```

Is it better to land on the `if(FALSE)` line in this case? Then it's immediately clear from the lint message what's triggering the lint. In most IDEs, it will also land on the line with the 'collapse' arrow ⬇️ to more easily highlight all the relevant code.

**Comment by IndrajeetPatil** [comment]
With #2391:

``` r
library(lintr)

lint("if (FALSE) {
  foo()
}",
  unreachable_code_linter()
)
#> <text>:1:1: warning: [unreachable_code_linter] Code inside a conditional loop with a deterministically false condition should be removed.
#> if (FALSE) {
#> ^~~~~~~~~~~~
```

<sup>Created on 2023-12-05 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>
